server/src/controller.ts

```diff
@@ -21,12 +34,20 @@ export let registration = async (req: Request, res: Response) => {
             return;
         }
 
-        await User.create({ username: req.body.username, password: req.body.password }, function (err: any, user: any) {
-            if (err) {
-                res.status(501).send("Error!");
+        await User.findOne({username: req.body.username}, (err: any, user: any) => {
+            if (!user) {
+                User.create({ username: req.body.username, password: req.body.password },
+                        function (err: any, user: any) {
+                    if (err) {
+                        res.status(501).send("Error!");
+                    } else {
+                        res.send();

```

Comment:
why is empty?

Reply:
What should I return to the front-end? Status code 200 is enough to know the registration is successful.

Reply:
I believe it is fine like that